app/src/main/java/org/oppia/android/app/settings/profile/SnackbarManager.kt

```diff
@@ -0,0 +1,70 @@
+package org.oppia.android.app.settings.profile
+
+import android.util.Log
+import android.view.View
+import androidx.annotation.StringRes
+import androidx.appcompat.app.AppCompatActivity
+import com.google.android.material.snackbar.Snackbar
+import org.oppia.android.domain.snackbar.SnackbarController
+import org.oppia.android.util.data.AsyncResult
+import org.oppia.android.util.data.DataProviders.Companion.toLiveData
+import javax.inject.Inject
+
+class SnackbarManager @Inject constructor(private val snackbarController: SnackbarController) {
+
+  fun showSnackbar(@StringRes messageStringId: Int, duration: SnackbarController.SnackbarDuration) {
+    snackbarController.enqueueSnackbar(
+      SnackbarController.SnackbarRequest.ShowSnackbar(
+        messageStringId,
+        duration
+      )
+    )
+  }
+
+  fun enableShowingSnackbars(activity: AppCompatActivity) {
+    snackbarController.getCurrentSnackbar().toLiveData().observe(activity) { result ->
+      when (result) {
+        is AsyncResult.Success -> when (val request = result.value) {
+          is SnackbarController.SnackbarRequest.ShowSnackbar -> showSnackbar(
+            activity.findViewById(
+              android.R.id.content
+            ),
+            request
+          )
+          SnackbarController.SnackbarRequest.ShowNothing -> {
+            if (snackbarController.snackbarRequestQueue.isNotEmpty()) {
+              snackbarController.notifyPotentialSnackbarChange()
```

Comment:
No need for this call since it is called  in ` snackbarController.dismissCurrentSnackbar()`, making it redundant here.

Reply:
okay